aws/src/main/java/org/apache/iceberg/aws/glue/DynamoLockManager.java

```diff
@@ -0,0 +1,190 @@
+          Thread.currentThread().interrupt();
+          LOG.warn("Glue lock DynamoDB table creation sleep interrupted", ie);
+        }
+        try {
+          DescribeTableResponse describeTableResponse = dynamo.describeTable(DescribeTableRequest.builder()
+              .tableName(tableName)
+              .build());
+          isTableActive = describeTableResponse.table().tableStatus().equals(TableStatus.ACTIVE);
+        } catch (ResourceNotFoundException e2) {
+          LOG.error("describe table failed after table creation", e2);
+        }
+      }
+
+      if (!isTableActive) {
+        throw new IllegalStateException(String.format("DynamoDB table <%s> failed to become active after %d ms",
+            tableName, LOCK_TABLE_CREATION_WAIT_MS * LOCK_TABLE_CREATION_WAIT_ROUNDS));
+      }
+    }
+  }
+
+
+  @Override
+  public boolean tryLock(String lockId, long expireMillis) {
+
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put(LOCK_TABLE_COL_LOCK_ID, AttributeValue.builder().s(lockId).build());
+    // TODO: fix potential clock skew issue
+    //   more details: https://github.com/awslabs/amazon-dynamodb-lock-client#how-we-handle-clock-skew
+    key.put(LOCK_TABLE_COL_EXPIRE_TS_MILLIS, AttributeValue.builder().n(
+        Long.toString(System.currentTimeMillis() + expireMillis)
+    ).build());
+
+    Map<String, AttributeValue> expressionValues = Maps.newHashMap();
+    expressionValues.put(":lid", AttributeValue.builder().s(lockId).build());
+    expressionValues.put(":ts", AttributeValue.builder().n(Long.toString(System.currentTimeMillis())).build());
+
+    try {
+
+      dynamo.putItem(PutItemRequest.builder()
+          .tableName(awsProperties.glueCatalogLockTable())
+          .item(key)
+          // succeed only if there is no lock, or the lock is there but already expired
+          .conditionExpression("attribute_not_exists(" +
+              LOCK_TABLE_COL_LOCK_ID + ") OR (" +
+              LOCK_TABLE_COL_LOCK_ID + " = :lid AND " +
+              LOCK_TABLE_COL_EXPIRE_TS_MILLIS + " < :ts)")
+          .expressionAttributeValues(expressionValues)
+          .build());
+      return true;
+    } catch (ConditionalCheckFailedException e) {
+      LOG.debug("Acquiring lock {} failed", lockId, e);
+      return false;
+    } catch (Exception e) {
+      LOG.error("Encountered unexpected exception when acquiring lock {}", lockId, e);
+      return false;
+    }
+  }
+
+  @Override
+  public void unlock(String lockId) {
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put(LOCK_TABLE_COL_LOCK_ID, AttributeValue.builder().s(lockId).build());
+
+    dynamo.deleteItem(DeleteItemRequest.builder()
+        .tableName(awsProperties.glueCatalogLockTable())
+        .key(key)
+        .build());
+  }
+
+  /**
+   * The lock table schema, for users who would like to create the table separately
+   * @return lock table schema
+   */
+  public static List<KeySchemaElement> getLockTableSchema() {
```

Comment:
Iceberg generally avoids using "get" in names unless it is making the API more clear and understandable. Here, it is sufficient to use `lockTableSchema()` and `lockTableColumnDefinitions()`